spec/unit/lib/cloud_controller/diego/docker/staging_completion_handler_spec.rb

```diff
@@ -43,6 +43,96 @@ module Docker
             expect(runner).to have_received(:start)
           end
 
+          context 'when it receives lifecycle_data in response' do
+            let(:payload) do
+              {
+                execution_metadata: '"{\"cmd\":[\"start\"]}"',
+                detected_start_command: { web: 'start' },
+                lifecycle_data: { docker_image: docker_image_name }
+              }
+            end
+            let(:droplet) { app.reload.current_droplet }
+
+            context 'with cached image' do
+              let(:docker_image_name) { '10.244.2.6:8080/generated_id:latest' }
+
+              it 'updates the app with the new image' do
+                handler.staging_complete(staging_guid, payload)
+
+                expect(droplet.cached_docker_image).to eq(docker_image_name)
+              end
+            end
+
+            context 'with empty cached image' do
+              let(:docker_image_name) { '' }
+
+              it 'does not update the cached image' do
+                handler.staging_complete(staging_guid, payload)
+
+                expect(droplet.cached_docker_image).to be_nil
+              end
+            end
+          end
+
+          shared_examples 'a handler with no lifecycle data' do
```

Comment:
I'd question why creating a shared example when it seems (from this PR) being used only twice...  

I say this because, shared example have caused a lot of "pain" in services team that the policy has been to avoid them. I'd say especially in a situation when the shared example is not shared so many times...  

Finally, if you still decide to use shared example, might be good to aggregate in one place.

Anyhow, my .02